android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -476,7 +493,7 @@ private void updateDebugView(PwPair pwPair, View view) {
 
     public void clear() {
       mPwPairs.clear();
-      notifyDataSetChanged();
+      notifyChangeOnUiThread();
```

Comment:
I'd rather we called clear() on the UI thread in calling methods.  Also, please reserve this for a distinct change.